backend/services/tool/tool_service.py

```
@@ -0,0 +1,380 @@
+            tool_name: Name of the tool
+            repo_name: Name of the repository
+            user_id: ID of the user
+            
+        Raises:
+            NotFoundException: If tool not found
+        """
+        tool_path = self._get_tool_path(tool_name, repo_name, user_id)
+        
+        if not tool_path.exists():
+            logger.warning(f"Tool file not found for deletion: {tool_path}")
+            raise NotFoundException(f"Tool '{tool_name}' not found in repository '{repo_name}'")
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)